.github/workflows/master-push.yml

```diff
@@ -0,0 +1,87 @@
+name: Push to master
+
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
```

Comment:
Why use macos on master, but not PR? If there are really problems with macos, they would only show after the merge.

Reply:
Just to save some time.  I don't think the very minor differences from linux (and nothing in our code we know of) are worth running on each of the possibly many pushes in one PR before it is merged.  My philosophy was to keep CI time down to a minimum while catching 99.x% of errors at the push stage, and the other 100-.x at merge.  With just two OSs and three Python versions + linter, we can run almost all in parallel with up to 6 parallel sessions, or at least have 4 parallel x 2 sequential.  And the sphinx build is still coming...

I'm guessing the merge will be aborted if there are unsuccessful steps, but I guess we won't know until we see a case or force an error somehow.  In the worst case I'll get an email saying the build was unsuccessful, and we can open the PR again.

Reply:
Fair enough, sounds good!